cerberus-api-tests/src/main/groovy/com/nike/cerberus/api/CerberusApiActions.groovy

```diff
@@ -93,13 +104,19 @@ class CerberusApiActions {
     }
 
     static def getDecryptedPayload(String iamPrincipalArn, String region, String base64EncodedKmsEncryptedAuthPayload, boolean assumeRole = true) {
-        AWSKMSClient kmsClient
-        if (assumeRole) {
-            kmsClient = new AWSKMSClient(new STSProfileCredentialsServiceProvider(
-                    new RoleInfo().withRoleArn(iamPrincipalArn)
-                            .withRoleSessionName(UUID.randomUUID().toString()))).withRegion(Regions.fromName(region))
-        } else {
-            kmsClient = new AWSKMSClient().withRegion(Regions.fromName(region))
+        AWSKMSClient kmsClient = kmsClientCache.getIfPresent(new Tuple2(iamPrincipalArn, region))
```

Comment:
Does `getIfPresent` do the right thing in terms of equality (e.g. String comparison, and comparing both values in the tuple? I assume so..?